peps/pep-0752.rst

```diff
@@ -208,78 +169,56 @@ Semantics
-a hyphen to the end of the proposed and existing namespace. An overlap is
-detected when any existing namespace starts with the proposed namespace.
+repositories. For example, if PyPI has a namespace ``acme`` that is owned by
+the company Acme, packages starting with ``acme-`` that come from other
+non-PyPI mirror repositories do not confer the same level of trust.
+
+Grants MUST NOT overlap ownership. For example, if there is an existing grant
+for ``foo-bar`` then a new grant for ``foo`` would only be possible for the
+owner of the former. An overlap is determined by comparing the
+`normalized <naming_>`_ proposed namespace with the normalized namespace of
+every existing root grant. Every comparison must append a hyphen to the end of
+the proposed and existing namespace. An overlap is detected when any existing
+namespace starts with the proposed namespace.
+
+Repositories SHOULD impose a depth limit on the number of hyphens in a namespace.
+For example, if the depth limit is ``1`` then the namespace ``foo-bar`` would be
+allowed but ``foo-bar-baz`` could not be granted.
 
 .. _uploads:
 
 Uploads
 -------
 
-If the name of a package being uploaded matches a reserved namespace and either
-of the following criteria are true:
-
-* The project does not yet exist.
-* The project is not owned by an organization with an active grant for the
-  namespace.
-
-Then the upload MUST fail with a 403 HTTP status code.
-
-.. _open-namespaces:
-
-Open Namespaces
------------------
-
-The owner of a grant may choose to allow others the ability to release new
-projects with the associated namespace. Doing so MUST allow
-`uploads <uploads_>`_ for new projects matching the namespace from any user.
-
-It is possible for the owner of a namespace to both make it open and allow
-other organizations to use the grant. In this case, the authorized
-organizations have no special permissions and are equivalent to an open grant
-without ownership.
+Uploads MUST fail with a :rfc:`409 Conflict <9110#name-409-conflict>` HTTP
+status code if the name of a package being uploaded matches a reserved namespace
+and the project owner does not have an active grant for the namespace.
 
-.. _hidden-grants:
-
-Hidden Grants
--------------
-
-Repositories MAY create hidden grants that are not visible to the public which
-prevent their namespaces from being claimed by others. Such grants MUST NOT be
-`open <open-namespaces_>`_ and SHOULD NOT be exposed in the
-`API <repository-metadata_>`_.
-
-Hidden grants are useful for repositories that wish to enforce upload
-restrictions without the need to expose the namespace to the public.
+Repositories SHOULD have an exception to this rule for projects that existed
+before the namespace was reserved.
 
 .. _repository-metadata:
 
 Repository Metadata
 -------------------
 
-The :pep:`JSON API <691>` version will be incremented from ``1.2`` to ``1.3``.
+The :pep:`JSON API <691>` version will be incremented from ``1.3`` to ``1.4``.
```

Comment:
```suggestion
The :pep:`JSON API <691>` version will be incremented from ``1.4`` to ``1.5``.
```
Since [PEP 792](https://peps.python.org/pep-0792/) has since been accepted which uses version `1.4`.